src/api.d.ts

```diff
@@ -223,3 +224,33 @@ export declare function registerApplicationResourceProvider(id: string, provider
 // resource extensions need to activate onView:localResourceView and call this
 export declare function registerLocalResourceProvider(id: string, provider: LocalResourceProvider): vscode.Disposable;
 
+export interface ActivityTreeItemOptions {
+    label: string;
+    contextValuePostfix?: string;
+    collapsibleState?: vscode.TreeItemCollapsibleState;
+    children?: (parent: AzExtParentTreeItem) => AzExtTreeItem[];
+}
+
+interface ActivityType {
+    inital(): ActivityTreeItemOptions;
+    onSuccess(): ActivityTreeItemOptions;
+    onError(error: IParsedError): ActivityTreeItemOptions;
+}
+
+export declare abstract class ActivityBase<R = void> implements ActivityType {
+
+    abstract inital(): ActivityTreeItemOptions;
+    abstract onSuccess(): ActivityTreeItemOptions;
+    abstract onError(error: IParsedError): ActivityTreeItemOptions;
+
+    public done: boolean;
+    public error?: IParsedError;
+    public readonly task: () => Promise<R | undefined>;
+    public startedAtMs: number;
+
+    public constructor(task: () => Promise<R | undefined>);
+
+    public run(): Promise<R | undefined>;
+
+    public get state(): ActivityTreeItemOptions;
```

Comment:
Maybe this to be extra clear?
```suggestion
    public get currentState(): ActivityTreeItemOptions;
```